frontend/src/concepts/connectionTypes/utils.ts

```diff
@@ -121,56 +118,83 @@ export const S3ConnectionTypeKeys = [
   'AWS_S3_BUCKET',
 ];
 
-export enum ModelServingCompatibleConnectionTypes {
-  ModelServing = 's3',
-  OCI = 'oci-compliant-registry-v1',
-  URI = 'uri-v1',
+export enum ModelServingCompatibleTypes {
+  S3ObjectStorage = 'S3 compatible object storage',
+  OCI = 'OCI compliant registry',
+  URI = 'URI',
 }
 
-const modelServinConnectionTypes: Record<
-  ModelServingCompatibleConnectionTypes,
+const modelServingCompatibleTypesMetadata: Record<
+  ModelServingCompatibleTypes,
   {
     name: string;
+    resource: string;
     envVars: string[];
+    isConnectionCompatible?: (connection: Connection) => boolean;
   }
 > = {
-  [ModelServingCompatibleConnectionTypes.ModelServing]: {
-    name: 'S3 compatible object storage',
+  [ModelServingCompatibleTypes.S3ObjectStorage]: {
+    name: ModelServingCompatibleTypes.S3ObjectStorage,
+    resource: 's3',
     envVars: S3ConnectionTypeKeys,
+    isConnectionCompatible: (connection) =>
+      connection.metadata.annotations['opendatahub.io/connection-type'] === 's3' &&
```

Comment:
So this will not show model serving compatibility for connections with aws env vars but not the type of `s3` I believe? 
![image](https://github.com/user-attachments/assets/0772a78d-9d41-4f76-9e6e-1931841a92bf)

Reply:
Correct because the `type` makes it into the `storage-config` and is necessary for model serving to work.
That being said, if all env vars for s3 are populated then we would add the annotation on edit / save. Unless I missed a case.